platform/ios/src/MGLMapboxEvents.m

```diff
@@ -357,253 +306,198 @@ + (void)pauseMetricsCollection {
-    if (_locationManager || _paused) {
-        NSAssert(!(_locationManager && _paused),
+    if (self.locationManager || self.paused) {
+        NSAssert(!(self.locationManager && self.paused),
                  @"MGLMapboxEvents should not have a CLLocationManager while paused.");
         return;
     }
-    _locationManager = [[CLLocationManager alloc] init];
-    _locationManager.desiredAccuracy = kCLLocationAccuracyKilometer;
-    _locationManager.distanceFilter = 10;
-    _locationManager.delegate = self;
+    self.locationManager = [[CLLocationManager alloc] init];
+    self.locationManager.desiredAccuracy = kCLLocationAccuracyKilometer;
+    self.locationManager.distanceFilter = 10;
+    self.locationManager.delegate = self;
 
-    [_locationManager startUpdatingLocation];
+    [self.locationManager startUpdatingLocation];
 
     // -[CLLocationManager startMonitoringVisits] is only available in iOS 8+.
-    if ([_locationManager respondsToSelector:@selector(startMonitoringVisits)]) {
-        [_locationManager startMonitoringVisits];
+    if ([self.locationManager respondsToSelector:@selector(startMonitoringVisits)]) {
+        [self.locationManager startMonitoringVisits];
     }
 }
 
++ (void)pushEvent:(NSString *)event withAttributes:(MGLMapboxEventAttributes *)attributeDictionary {
+    [[MGLMapboxEvents sharedManager] pushEvent:event withAttributes:attributeDictionary];
+}
 
-- (void) pushTurnstileEvent {
+- (void)pushEvent:(NSString *)event withAttributes:(MGLMapboxEventAttributes *)attributeDictionary {
+    if (!event) {
+        return;
+    }
+    
+    if ([event isEqualToString:MGLEventTypeMapLoad]) {
+        [self pushTurnstileEvent];
+    }
+    
+    if ([self isPaused]) {
+        return;
+    }
+    
+    MGLMutableMapboxEventAttributes *evt = [MGLMutableMapboxEventAttributes dictionaryWithDictionary:attributeDictionary];
+    
+    // mapbox-events stock attributes
+    [evt setObject:event forKey:@"event"];
+    [evt setObject:@(version) forKey:@"version"];
+    [evt setObject:[self.rfc3339DateFormatter stringFromDate:[NSDate date]] forKey:@"created"];
+    [evt setObject:self.instanceID forKey:@"instance"];
+    [evt setObject:self.data.vendorId forKey:@"vendorId"];
+    [evt setObject:self.appBundleId forKey:@"appBundleId"];
+    
+    // mapbox-events-ios stock attributes
+    [evt setValue:self.data.model forKey:@"model"];
+    [evt setValue:self.data.iOSVersion forKey:@"operatingSystem"];
+    [evt setValue:@(self.data.scale) forKey:@"resolution"];
+    MGLReachability *reachability = [MGLReachability reachabilityForLocalWiFi];
+    [evt setValue:([reachability isReachableViaWiFi] ? @YES : @NO) forKey:@"wifi"];
+    [evt setValue:[self deviceOrientation] forKey:@"orientation"];
+    [evt setValue:@([self batteryLevel]) forKey:@"batteryLevel"];
+    [evt setValue:[self applicationState] forKey:@"applicationState"];
+    [evt setValue:@([self contentSizeScale]) forKey:@"accessibilityFontScale"];
```

Comment:
There’s inconsistent usage of `-setObject:forKey:` versus `-setValue:forKey:` here. Moreover, since all of the value-setting is unconditional, this should all be collapsed into a call to `-[NSMutableDictionary dictionaryWithObjectsAndKeys:]` and merged with `attributeDictionary` using `-addEntriesFromDictionary:`.